packages/iov-cosmos/src/decode.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Address,
+  Algorithm,
+  Amount,
+  ChainId,
+  Fee,
+  FullSignature,
+  Identity,
+  Nonce,
+  PubkeyBundle,
+  PubkeyBytes,
+  SendTransaction,
+  SignatureBytes,
+  SignedTransaction,
+  TokenTicker,
+} from "@iov/bcp";
+import { Encoding } from "@iov/encoding";
+import amino from "@tendermint/amino-js";
+
+import { AminoTx } from "./types";
+
+const { fromBase64 } = Encoding;
+
+export function decodePubkey(pubkey: amino.PubKey): PubkeyBundle {
+  return {
+    algo: Algorithm.Secp256k1,
+    data: fromBase64(pubkey.value) as PubkeyBytes,
+  };
+}
+
+export function decodeSignature(signature: string): SignatureBytes {
+  return fromBase64(signature) as SignatureBytes;
+}
+
+export function decodeFullSignature(signature: amino.StdSignature, nonce: number): FullSignature {
+  return {
+    nonce: nonce as Nonce,
+    pubkey: decodePubkey(signature.pub_key),
+    signature: decodeSignature(signature.signature),
+  };
+}
+
+export function decodeAmount(amount: amino.Coin): Amount {
+  return {
+    fractionalDigits: 9,
+    quantity: amount.amount,
+    tokenTicker: amount.denom as TokenTicker,
+  };
+}
+
+export function parseMsg(msgs: readonly amino.Msg[]): SendTransaction {
```

Comment:
Why does this function accept an array? We should have a length check at least of we want to process 1-element arrays.